client/src/store/stores/route.ts

```diff
@@ -42,7 +51,43 @@ export default class RouteStore {
     return pp;
   }
 
-  @action.bound private dropCache() {
+  @computed get namespace(): string | null {
+    return this.pathParts[0] || null;
+  }
+
+  @action.bound
+  navigate(to: string, options?: NavigateOptions<{}>) {
+    return this.history.navigate(to, options);
+  }
+
+  @action.bound
+  private dropCache() {
     this.cache = {};
   }
+
+  private listen() {
+    this.history.listen(({ location, action }) => {
+      // console.log('in history listen: ', location, action);
+      runInAction(() => {
+        this.dropCache();
+        this.location = location;
+      });
+    });
+
+    autorun(() => {
+      if (this.namespace) {
+        localStorage.setItem(RouteStore.NAMESPACE_LS_KEY, this.namespace);
+      }
+    });
+  }
+
+  private restoreNamespace() {
+    const currentNamespace = this.location.pathname.split('/').slice(1)[0];
+    if (!currentNamespace) {
```

Comment:
Consider inverting `if` condition to reduce indentation level and improve readability
```ts
const currentNamespace = this.location.pathname.split('/').slice(1)[0];
if (currentNamespace) return;

const storedNamespace = localStorage.getItem(RouteStore.NAMESPACE_LS_KEY);
if (!storedNamespace) return;

this.navigate(`/${storedNamespace}`);
```